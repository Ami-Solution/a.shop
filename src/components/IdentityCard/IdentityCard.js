import React, { PureComponent } from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import tokens from '../../styles/tokens';
import Button from '../Button';
import { H1 } from '../Headings';
import { Margin } from '../Spaces';
import grandmaIcon from '../../assets/home/icon_grand_mere.svg';

const Wrapper = styled.div`
  padding: ${tokens.spaces.m};
`;

const Card = styled.div`
  border-radius: ${tokens.radius.m};
  background: ${tokens.colors.white};
  box-shadow: 0 0 0.8rem rgba(0, 0, 0, 0.2);
  overflow: hidden;
`;

const UserProfile = styled.div`
  padding: 0 ${tokens.spaces.xs};
  flex: 1;
`;

const TopWrapper = styled.div`
  position: relative;
  display: flex;
  flex-flow: row nowrap;
  align-items: center;
  justify-content: center;
  text-align: left;
  padding: ${tokens.spaces.xs};
`;

const ButtonWrapper = Margin.extend`
  border-radius: ${tokens.radius.m};
  box-shadow: 0 0 0.8rem rgba(0, 0, 0, 0.2);
`;

class IdentityCard extends PureComponent {
  static propTypes = {
    children: PropTypes.node.isRequired,
    buttonLink: PropTypes.string,
    buttonText: PropTypes.string,
    cardName: PropTypes.string.isRequired
  };

  static defaultProps = {
    buttonLink: null,
    buttonText: null
  };

  render() {
    const { buttonLink, buttonText, cardName, children } = this.props;
    return (
      <Wrapper>
        <Card>
          <TopWrapper>
            <img src={grandmaIcon} alt="profile" width="65" />
            <UserProfile>
              <H1 light>{cardName}</H1>
            </UserProfile>
          </TopWrapper>
          {children}
        </Card>
        {!!buttonLink && (
          <ButtonWrapper top="m">
            <Button theme="primary" fullWidth href={buttonLink} target="blank">
              {buttonText}
            </Button>
          </ButtonWrapper>
        )}
      </Wrapper>
    );
  }
}

export default IdentityCard;
