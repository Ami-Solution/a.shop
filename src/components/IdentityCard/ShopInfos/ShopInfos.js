import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import tokens from '../../../styles/tokens';
import { H1 } from '../../Headings';

const BottomWrapper = styled.div`
  padding: ${tokens.spaces.s};
  background: ${tokens.colors.grey.lightest};
`;

const Adress = styled.p`
  color: ${tokens.colors.blue};
  margin-top: ${tokens.spaces.xxs};
  font-weight: 300;
`;

const Desc = styled.p`
  margin-top: ${tokens.spaces.xxs};
  font-size: ${tokens.fontSizes.xs};
  margin-bottom: 0;
  font-weight: 300;
`;

const ShopInfos = ({ title, adress, description }) => (
  <BottomWrapper>
    <H1>{title}</H1>
    <Adress>{adress}</Adress>
    <Desc>{description}</Desc>
  </BottomWrapper>
);

ShopInfos.propTypes = {
  title: PropTypes.string.isRequired,
  adress: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired
};

export default ShopInfos;
