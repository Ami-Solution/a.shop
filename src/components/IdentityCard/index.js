export { default as ShopInfos } from './ShopInfos';
export { default } from './IdentityCard';
