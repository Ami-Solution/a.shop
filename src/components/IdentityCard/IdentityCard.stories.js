/* eslint-disable */
import React from 'react'
import styled, { css } from 'styled-components'
import { storiesOf } from '@storybook/react'

import IdentityCard from './index';
import ShopInfos from './ShopInfos';

const props = {
  volumeTrade: 10.56,
  nbTrade: 4,
  amountToSell: 700,
  percentFees: 2.36,
  currency: "$"
}

storiesOf('03 - Components/IdentityCard', module)
.add('Shop IdentityCard', () => (
  <div style={{maxWidth: 340, margin: "0 auto"}}>
    <IdentityCard buttonLink="http://google.com" buttonText="Itinerary" cardName="Shop Name">
      <ShopInfos
        title="Welcome"
        adress="31 rue de Cotte 75012 Paris"
        description="Vente d'articles cuisine et dÃ©co pour toute la maison."
      />
    </IdentityCard>
  </div>
))
