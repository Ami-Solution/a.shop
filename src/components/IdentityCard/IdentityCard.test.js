/* global describe it expect */
import React from 'react';
import renderer from 'react-test-renderer';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import IdentityCard from './index';
import ShopInfos from './ShopInfos';

configure({ adapter: new Adapter() });

const props = {
  volumeTrade: 10.56,
  nbTrade: 4,
  amountToSell: 700,
  percentFees: 2.36,
  currency: "$"
}

describe('Component IdentityCard', () => {

  it('should be render shopCard without crash', () => {
    const component = renderer.create(
      <IdentityCard buttonLink="http://google.com" buttonText="Itinerary" cardName="Shop Name">
        <ShopInfos
          title="Welcome"
          adress="31 rue de Cotte 75012 Paris"
          description="Vente d'articles cuisine et dÃ©co pour toute la maison."
        />
      </IdentityCard>);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
