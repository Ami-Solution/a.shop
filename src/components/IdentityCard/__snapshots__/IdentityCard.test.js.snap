// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component IdentityCard should be render shopCard without crash 1`] = `
<div
  className="sc-fjdhpX bCqjUD"
>
  <div
    className="sc-jzJRlG iwTJuW"
  >
    <div
      className="sc-kAzzGY chxoco"
    >
      <img
        alt="profile"
        src="icon_grand_mere.svg"
        width="65"
      />
      <div
        className="sc-cSHVUG huGCJq"
      >
        <h1
          className="sc-htpNat hBugQt"
        >
          Shop Name
        </h1>
      </div>
    </div>
    <div
      className="sc-gzVnrw kjdDyw"
    >
      <h1
        className="sc-htpNat iRBdXH"
      >
        Welcome
      </h1>
      <p
        className="sc-htoDjs jQUsuP"
      >
        31 rue de Cotte 75012 Paris
      </p>
      <p
        className="sc-dnqmqq eAQBFl"
      >
        Vente d'articles cuisine et d√©co pour toute la maison.
      </p>
    </div>
  </div>
  <div
    className="sc-chPdSV cEFAOw"
  >
    <button
      className="sc-iwsKbI amyUb"
      disabled={false}
      onClick={[Function]}
    >
      Itinerary
    </button>
  </div>
</div>
`;
